compiler/resolution/functionResolution.cpp

```diff
@@ -4286,18 +4301,41 @@ static void generateUnresolvedMsg(CallInfo& info, Vec<FnSymbol*>& visibleFns) {
     int nPrintDetails = 1;
     int nPrint = 3;
 
+
+    // If there is more than one visible function and we haven't been
+    // asked to print all candidates, let's try to eliminate "worse"
+    // matches (e.g., method call / standalone function call
+    // mismatches).
```

Comment:
Are you sure we don't want to just adjust the sorting used above in `sortExampleCandidates` ? If you can arrange for the candidates in the case you care about to appear first in the sorted list, that might work better than filtering. It has slightly different behavior but it seems better IMO.

Reply:
In the original motivating example, the problem wasn't the order in which the unrelated `.write()` methods were printed, it was more that they were printed at all, which is what this PR is trying to improve.  So I don't see that adjusting the sort order would help without these other changes?

Reply:
That's fine; I haven't been following the motivating example in detail.

Reply:
It's in the OP if you're interested.  The idea is that none of the other 4 candidates for write() are printed because none of them are standalone as the function call itself was.